services/debugger/destination/eventDeliveryStatusUploader.go

```diff
@@ -135,3 +161,18 @@ func backendConfigSubscriber(backendConfig backendconfig.BackendConfig) {
 		updateConfig(config.Data.(backendconfig.ConfigT))
 	}
 }
+
+func recordHistoricEventsDelivery(destinationIDs []string) {
+	for _, destinationID := range destinationIDs {
+		var historicEventsDelivery []*DeliveryStatusT
+		eventsCacheMapLock.Lock()
+		if deliveryStatus, ok := eventsDeliveryCacheMap[destinationID]; ok {
+			historicEventsDelivery = deliveryStatus
+			delete(eventsDeliveryCacheMap, destinationID)
+		}
```

Comment:
I am not aware of the product requirements here, but should we delete them every time we send them ?

Reply:
Keep this in cache, don't delete it, if the control plane supports the polling for the events then we can send the events in case of let say a browser refresh.